server/src/test/java/io/crate/lucene/NestedArrayLuceneQueryBuilderTest.java

```diff
@@ -48,4 +48,11 @@ public void test_empty_nested_array_equals() {
         var query = convert("a = [[]]");
         assertThat(query.toString()).isEqualTo("+NumTermsPerDoc: a +(a = [[]])");
     }
+
+    @Test
+    public void test_any_equals_nested_array_literal() {
+        var query = convert("a = any([ [ [1], [1, 2] ], [ [3], [4, 5] ] ])");
+        // pre-filter by a terms query with 1, 2, 3, 4, 5 then a generic function query to make sure an exact match
+        assertThat(query.toString()).isEqualTo("+a:{1 2 3 4 5} #(a = ANY([[[1], [1, 2]], [[3], [4, 5]]]))");
```

Comment:
For example, inserting `[ [1], [1, 2] ]` will create the following indexes:
```
IntField(a,1)
IntField(a,1)
IntField(a,2)
```

prefiltering rows by `a` containing `1, 2, 3, 4, 5` followed by a generic function filter `(a = ANY([[[1], [1, 2]], [[3], [4, 5]]]))")` will be able to match `[ [1], [1, 2] ]`.